# HG changeset patch
# User Roman Neuhauser <neuhauser@sigpipe.cz>
# Date 1299595849 -3600
# Node ID bc8c9d787e8e7ab5b9d87b360edb5eaba728181b
# Parent  18aaaefb803c4954dc1c9a72bab93ee5459b5010
(crashing) test for double close() after archive_read_open_fd

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -250,6 +250,7 @@
 	libarchive/test/test_archive_api_feature.c		\
 	libarchive/test/test_archive_clear_error.c		\
 	libarchive/test/test_archive_read_close_twice.c		\
+	libarchive/test/test_archive_read_close_twice_open_fd.c	\
 	libarchive/test/test_archive_read_close_twice_open_filename.c	\
 	libarchive/test/test_archive_set_error.c		\
 	libarchive/test/test_archive_read_set_filter_option.c	\
diff --git a/libarchive/test/CMakeLists.txt b/libarchive/test/CMakeLists.txt
--- a/libarchive/test/CMakeLists.txt
+++ b/libarchive/test/CMakeLists.txt
@@ -15,6 +15,7 @@
     test_archive_api_feature.c
     test_archive_clear_error.c
     test_archive_read_close_twice.c
+    test_archive_read_close_twice_open_fd.c
     test_archive_read_close_twice_open_filename.c
     test_archive_set_error.c
     test_archive_read_set_filter_option.c
diff --git a/libarchive/test/test_archive_read_close_twice_open_filename.c b/libarchive/test/test_archive_read_close_twice_open_fd.c
copy from libarchive/test/test_archive_read_close_twice_open_filename.c
copy to libarchive/test/test_archive_read_close_twice_open_fd.c
--- a/libarchive/test/test_archive_read_close_twice_open_filename.c
+++ b/libarchive/test/test_archive_read_close_twice_open_fd.c
@@ -37,11 +37,11 @@
 assertEqualInt(0, archive_errno(a)); \
 assertEqualString(NULL, archive_error_string(a));
 
-DEFINE_TEST(test_archive_read_close_twice_open_filename)
+DEFINE_TEST(test_archive_read_close_twice_open_fd)
 {
 	struct archive* a = archive_read_new();
 
-	assertEqualInt(0, archive_read_open_filename(a, NULL, 0));
+	assertEqualInt(0, archive_read_open_fd(a, 0, 0));
 	test(a);
 
 	assertEqualInt(0, archive_read_close(a));
